responsibleai_text/responsibleai_text/managers/explainer_manager.py

```diff
@@ -251,21 +263,25 @@ def _compute_text_feature_importances(self, explanation):
         """
         text_feature_importances = []
         is_classif_task = self._is_classification_task
-        for instance in explanation:
-            text_feature_importance = TextFeatureImportance()
-            if is_classif_task:
+        if is_classif_task:
+            for instance in explanation:
+                text_feature_importance = TextFeatureImportance()
                 text_feature_importance.localExplanations = \
                     instance.values.tolist()
                 text_feature_importance.text = instance.data
-            elif self._task_type == ModelTask.QUESTION_ANSWERING:
+                text_feature_importances.append(text_feature_importance)
+        elif self._task_type == ModelTask.QUESTION_ANSWERING:
+            for i_start, i_end in zip(explanation[0], explanation[1]):
                 text_feature_importance = TextFeatureImportance()
                 text_feature_importance.localExplanations = \
-                    instance.values.tolist()
-                text_feature_importance.text = instance.data
-            else:
-                raise ValueError("Unknown task type: {}".format(
-                    self._task_type))
-            text_feature_importances.append(text_feature_importance)
+                    [i_start.values.tolist(), i_end.values.tolist()]
+                text_feature_importance.text = i_start.data
+                text_feature_importance.baseValues = \
+                    [i_start.base_values.tolist(), i_end.base_values.tolist()]
+                text_feature_importances.append(text_feature_importance)
```

Comment:
For my knowledge: why is this last append needed?

Reply:
text_feature_importance contains an explanation for each document and text_feature_importances, with the extra s, contains the explanations for all documents.  So the append is called for each additional document or text phrase/sentence we are running on.